providers/flagd/pkg/provider.go

```diff
@@ -1,213 +1,164 @@
 package flagd
 
 import (
-	schemaV1 "buf.build/gen/go/open-feature/flagd/protocolbuffers/go/schema/v1"
 	"context"
-	"errors"
-	"fmt"
 	"github.com/go-logr/logr"
-	lru "github.com/hashicorp/golang-lru/v2"
-	flagdModels "github.com/open-feature/flagd/core/pkg/model"
-	flagdService "github.com/open-feature/flagd/core/pkg/service"
 	"github.com/open-feature/go-sdk-contrib/providers/flagd/internal/logger"
 	"github.com/open-feature/go-sdk-contrib/providers/flagd/pkg/cache"
-	"github.com/open-feature/go-sdk-contrib/providers/flagd/pkg/constant"
 	"github.com/open-feature/go-sdk-contrib/providers/flagd/pkg/service"
 	of "github.com/open-feature/go-sdk/pkg/openfeature"
-	"os"
-	"strconv"
-)
-
-// naming and defaults follow https://github.com/open-feature/flagd/blob/main/docs/other_resources/creating_providers.md?plain=1#L117
-const (
-	defaultMaxCacheSize                               int    = 1000
-	defaultPort                                              = 8013
-	defaultMaxEventStreamRetries                             = 5
-	defaultTLS                                        bool   = false
-	defaultCache                                      string = "lru"
-	defaultHost                                              = "localhost"
-	flagdHostEnvironmentVariableName                         = "FLAGD_HOST"
-	flagdPortEnvironmentVariableName                         = "FLAGD_PORT"
-	flagdTLSEnvironmentVariableName                          = "FLAGD_TLS"
-	flagdSocketPathEnvironmentVariableName                   = "FLAGD_SOCKET_PATH"
-	flagdServerCertPathEnvironmentVariableName               = "FLAGD_SERVER_CERT_PATH"
-	flagdCacheEnvironmentVariableName                        = "FLAGD_CACHE"
-	flagdMaxCacheSizeEnvironmentVariableName                 = "FLAGD_MAX_CACHE_SIZE"
-	flagdMaxEventStreamRetriesEnvironmentVariableName        = "FLAGD_MAX_EVENT_STREAM_RETRIES"
-	cacheDisabledValue                                       = "disabled"
-	cacheLRUValue                                            = "lru"
 )
 
 type Provider struct {
-	ctx                              context.Context
-	service                          service.IService
-	cacheEnabled                     bool
-	cache                            Cache[string, interface{}]
-	maxCacheSize                     int
-	providerConfiguration            *ProviderConfiguration
-	eventStreamConnectionMaxAttempts int
-	isReady                          chan struct{}
-	logger                           logr.Logger
-	otelIntercept                    bool
-}
-type ProviderConfiguration struct {
-	Port            uint16
-	Host            string
-	CertificatePath string
-	SocketPath      string
-	TLSEnabled      bool
-}
+	logger                logr.Logger
+	providerConfiguration *providerConfiguration
+	service               IService
+	status                of.State
 
-type ProviderOption func(*Provider)
+	eventStream chan of.Event
```

Comment:
channels "feel" great for this.

Reply:
Yeah, emit the event and let SDK handle it. IMO this is much better than shared objects + locks